File: Java/Remove Element.java
class Solution {

    /**
     * Time:   O(n)
     * Memory: O(1)
     */
    public int removeElement(int[] nums, int val) {
        int i = 0;

        for (int num : nums)
            if (num != val)
                nums[i++] = num;

        return i;
    }
}

File: Python/Remove Element.py
from typing import List


class Solution:
    """
    Time:   O(n)
    Memory: O(1)
    """

    def removeElement(self, nums: List[int], val: int) -> int:
        i = 0

        for num in nums:
            if num == val:
                continue
            nums[i] = num
            i += 1

        return i
